streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -741,6 +757,23 @@ public void close(boolean clean,
      * @param records   the records
      */
     public void addRecords(final TopicPartition partition, final Iterable<ConsumerRecord<byte[], byte[]>> records) {
+        // if condition put here in case of restarts and rebalances to check for correct timestamp
```

Comment:
We call `addRecords` during regular processing but only need to restore the partition time if a rebalance happened. Hence, I don't think this is the right place to add this code.

From my understanding, we should do this in `StreamThread#RebalanceListener.onPartitionsAssigned` instead? \cc @guozhangwang to confirm/comment

Reply:
Oh, thats a good suggestion. Would need to change test location in that case.

Reply:
I sync with Guozhang about this. It's not correct to get the committed offsets within `onPartitionsAssigned()` callback, because the consumer might not have fetched/updated the offsets from the brokers.

However, we should still move the off the main loop. We only need to initialize the partition time if we get a new task assigned. Hence, we should move it into `AssignedTasks#initializeNewTasks()`?